libafl_frida/src/asan/hook_funcs.rs

```diff
@@ -1608,7 +1608,7 @@ impl AsanRuntime {
     ) -> *mut c_void {
         log::trace!("hook_mmap");
         let res = original(addr, length, prot, flags, fd, offset);
-        if res != (-1_isize as *mut c_void) {
+        if !ptr::eq(res, -1_isize as *mut c_void) {
```

Comment:
This should very likely be addr_eq, I assume -1 can never be ptr::eq https://doc.rust-lang.org/std/ptr/fn.addr_eq.html

Reply:
Opened https://github.com/AFLplusplus/LibAFL/pull/3105 for that